libsoftwarecontainer/src/softwarecontainer.cpp

```diff
@@ -197,6 +197,23 @@ ReturnCode SoftwareContainer::startGateways(const GatewayConfiguration &gwConfig
 
 ReturnCode SoftwareContainer::configureGateways(const GatewayConfiguration &gwConfig)
 {
+    assertValidState();
+
+    // Make sure that all the gateway ids in the given GatewayConfig can map
+    // to a gateway in SoftwareContainer
+    for (auto &id : gwConfig.ids()) {
+        bool match = false;
+        for (auto &gateway : m_gateways) {
+            if (id == gateway->id()) {
+                match = true;
+            }
+        }
+        if (!match) {
+            throw GatewayError("Could not find any gateway matching id: " + id);
```

Comment:
Maybe this is a really small possibility but if there is no enabled gateway this could create false throw. Maybe it would be reasonable to check if there is any member in `m_gateways`.